# Getting Started {#Start}

So you tested your microbiome and now you want to know what it all means. Later we'll go into much more detail about the sampling process itself: differences between labs, how to get the best sample, how long it will take, etc. but let's start by assuming you've gone through all that. What will you learn?

## Using the uBiome web interface

You just received an email that uBiome has finished processing your sample. Now what do you do?

The first stop is the [uBiome Explorer web dashboard](http://explorer.ubiome.com/), where you'll see an overview of your results. The dashboard is intended to be self-explanatory, so you should feel free to click around to try the various options. The beginning page will look something like this:


If you only have one sample, your page will look something like Figure \@ref(fig:explorerStartPage).

```{r explorerStartPage, fig.cap = "uBiome Explorer Insights Overview"}
knitr::include_graphics(file.path(getwd(),"assets/ubiomeExplorerStartPageMar29Sample.jpg"))

```

The colorful pie chart there is your first high-level glimpse of the bacteria found in your sample. Hover over it with your mouse (or tap with your finger if you’re on a phone or iPad), to see a pop-up view of the approximate percentages.

You’re seeing a _phyla_-based view of your microbiome. As [we'll learn in more detail later](#science), this is a very high level summary, a bit like how a Martian might see life on earth from far in outer space. Keep in mind that by “very high level”, we mean _very_ high, the equivalent of seeing life only at the level of “vertebrate” and “invertebrate”. The incomprehensible diversity of bacterial life means that at this level, you don’t even bother separating fish from one another or for that matter, from land animals: just lump all of them together. At the phyla level, mammals, reptiles, fish – they’re all the same. Similarly, when you see the phylum _Firmicutes_, for example, think of it as having as encompassing as much variation as all the vertebrates, from those on land, in the ocean, big, small, harmless, dangerous and everything in between. In other words, this first approximation is pretty crude.

Note that the various phyla are represented as percentages of the whole, so some of them are relegated to tiny slices of the pie. That doesn’t necessarily mean they’re unimportant, though, because in the world of life, even some tiny populations can exert an outsized influence. A wolf is just one of a gazillion different creatures in Yellowstone National Park, but that is little comfort to a rabbit.

Remember too that so far we're treating each individual organism as if it has the same effect as the others, another oversimplification, just like it would be a mistake to assume that elephants matter more because they’re bigger than humans.

All we know at the phylum level is that, proportionately, certain large classes of bacteria are more well-represented than others. That may not sound like much to go on, but as we’ll see, scientists have been able to learn quite a bit from tiny clues like this.

Click next on the "Insights" tab ( Figure \@ref(fig:explorerInsightsPage)) which may be more informative for beginners.If this is your first time here, go ahead and breeze through this page, but just ignore most of the conclusions. As we'll learn, scientists have barely scratched the surface of what we will eventually understand about the microbiome, so think of this page as a teaser -- a hint of the potential. If your results show something you didn't expect, say on diversity or body weight, don't worry: it almost certainly just means that the Insights pane is wrong about you. Later we'll see how to pull out far more interesting and useful results.

```{r explorerInsightsPage, fig.cap = "The Insights tab provides a high-level view of what uBiome thinks is most actionable about your microbiome."}
knitr::include_graphics(file.path(getwd(),"assets/ubiomeExplorerInsightsPageMar29.jpg"))
```

The one part of the Insights page that you _should_ take seriously is the section called "most uncommon bacteria". Let's jump to that section now.

## How are you different?

Since so little about the microbiome is well-understood, most of the time it will be hard to interpret your results by themselves. This is where the power of uBiome's citizen science can help, because the web page lets you easily compare your results with other people.  

```{r explorerCompare, fig.cap = "The Compare view shows the differences between my sample and others in the uBiome database."}
knitr::include_graphics(file.path(getwd(),"assets/ubiomeExplorerCompareMar29.jpg"))
```


In this case, I can see that my sample has about 64% of the abundance of _Firmicutes_ as other people in the uBiome database. You can sort this field by abundance (ratio) and click on the various taxonomic categories to dig deeper into your microbiome and find microbes that are particularly rare in your sample. (Figure \@ref(fig:explorerRare))

```{r explorerRare, fig.cap = "My sample shows very low levels of these microbes, compared to the hundreds of thousands of others in the uBiome database."}
knitr::include_graphics(file.path(getwd(),"assets/ubiomeExploreRareMar29.jpg"))
```

For example, I found that I have much less _Prevotella_ and _Butyvibrio_, two common microbes than the vast majority of uBiome customers.  I also found much higher-than-normal levels of _Deinococcus-Thermus_, an extremophile usually only found in harsh environments, like the impossibly acidic human stomach. Why would I have so many more than other people like me?

One clue is the similarly-high levels of Proteobacteria, which as we just noted is often high in people with gastro-intestinal issues. Maybe I was exposed to something that released a bit of stomach acid, and its extremophile _Deinococcus_ into my colon.

_Verrucomicrobia_ is elevated too, though not by much. This is a Phylum that is generally associated with soil microbes, though it's not unusual to find it in the gut as well.

I looked back at my notes during the period of a few days before this sample, and interestingly I had been doing work in my garden at the time. We have deer and other animals that come near our house...is it possible that I was exposed to some animal excrement, maybe some of which somehow made it into my gut?

It's just a theory, and perhaps an unlikely one, but I wonder if my body was fighting some type of pathogen as a result of my outdoor activities. Now before we get too excited, keep in mind that I never felt the slightest bit ill during this period. Can I thank my microbiome? Perhaps I was exposed to something nasty, the kind of bug that under other circumstances might have wreaked havoc and left me terribly sick for days. Now I have no way of knowing for sure, but this tidbit makes me wonder if maybe my microbiome acted as a guardian angel, protecting me from something much worse.

Similarly, your own results comparing to others may lead you to clues about aspects of your environment you didn't otherwise suspect.

Incidentally, since this test I've sent dozens of additional samples to uBiome and never saw that _Deinococcus_ thing again. Good riddance? or maybe I should be thankful for the protection, and hope that little bits of it may still be present, undetectably in my stomach, waiting to fight another external enemy the next time I come in contact with something in the soil.

---

[^1]: De Filippo, Carlotta, Duccio Cavalieri, Monica Di Paola, Matteo Ramazzotti, Jean Baptiste Poullet, Sebastien Massart, Silvia Collini, Giuseppe Pieraccini, and Paolo Lionetti. 2010. “Impact of Diet in Shaping Gut Microbiota Revealed by a Comparative Study in Children from Europe and Rural Africa.” _Proceedings of the National Academy of Sciences of the United States of America_ 107 \(33\): 14691–96. doi:10.1073/pnas.1005963107.

## Diversity

The general consensus is that diversity is good: a greater variety of microbes ensures more resillience against the daily threat of invaders. Many people, after taking just one test, often feel either reassured that their diversity is "good" or disappointed that it's "bad". But I find that day-to-day variability is great enough that it's almost never useful to use a single result. For example, here's my diversity during a typical week:


```{r, plotOneWeekDiversity, echo= FALSE, warning = FALSE}
plot_richness(subset_samples(sprague.family.norm, Date > "2017-03-15" & Date < "2017-03-27" & Site == "gut"), measures=c("InvSimpson"),x = "Date") +
  scale_x_date(date_labels = "%m-%d (%A)", date_breaks = "1 day", date_minor_breaks = "1 day") +
  geom_smooth(method = "loess") + 
  ggtitle("Gut diversity changes day-to-day")
```

If I had only tested on Monday, I may have been disappointed with my 0.38 score. Wait another day or two and, with no significant changes in diet, I was up to 0.72 -- before plunging to 0.37 by the weekend. Moral: don't take a single result too seriously.

To get a sense of how much diversity can vary over a year:

```{r, plotOneYearDiversity, echo= FALSE, warning = FALSE}
plot_richness(subset_samples(sprague.family.norm, Date > "2016-06-01" & Site == "gut" & Reads > 10000), measures=c("InvSimpson"), x = "Date") +
  scale_x_date(date_labels = "%m-%Y", date_breaks = "3 months", date_minor_breaks = "1 month") +
  geom_smooth(method = "loess") + 
  ggtitle(paste("Gut diversity year-to-year through",
                tail(sample_data(gut.norm)$Date,1)))
  
```

Although the blue moving-average line shows apparent stability, there are many days that are far above and below the average. Yes, at various times during this period I was eating different types of food, often in a deliberate attempt to influence my microbiome, but believe me: that is not the reason for the wild changes up and down. I also studied the mathematics behind how diversity is measured, hoping to find something more "accurate", but ultimately I concluded that, like many attempts to summarize the microbiome in a single number, the whole concept of diversity is a mirage. Everything depends.


_Bifidobacterium_ is a key component of virtually all popular probiotic supplements, partly because it is so easy to manufacture, but also due to its proven association with sleep and other aspects of health. A six month picture of my levels shows some dramatic ups and downs^[Important note: in all of these charts, I don't distinguish between a day when the tested abundance was zero, and a day when I have no reliable test results. Besides the daily variability in the microbiome, the test results themselves often come with wild differences in quality. This is nothing to complain about: the ability to see these microbes in the first place is a true miracle of technology, and even the most careful lab can't get around the myriad of technical issues that come with such a complex process.
].

```{r summarPlotBifido}
#mhg_taxa_plot(subset_samples(gut.norm, Date > "2016-06-01"),"Akkermansia")

probiotic.day <- seq(as.Date("2016-10-23"),length=9, by=1)
plot_bar(subset_taxa(subset_samples(gut.norm, Date %in% seq(as.Date("2016-07-01"), length = 180, by = 1)),
                     Genus == "Bifidobacterium"), x = "Date") + 
  geom_smooth(method = "loess", se = FALSE) +
  annotate("point",x=probiotic.day, y = -0.03, color = "red")+
  scale_y_continuous(labels = function(x) x / 10000) + ylab("Abundance (%)") + ggtitle("Bifidobacterium")
  
```

Incidentally, the red dots indicate days when I was taking a powerful probiotic supplement that contained _Bifidobacterium_. And that big spike in September? That was during a trip to New Orleans, when I ate a lot of red beans and rice.  At least for me, food seems to work better than taking supplements.
