# Microbes to Watch {#Microbes}


Your gut as seen by consumer-priced sequencing technology contains many more unique microbial species than you can possibly track, at least hundreds in most people and potentially over 1000. I've seen 1083 different ones in my own results. And that's just using the comparatively crude 16S technology: more comprehensive estimates based on other technology find as many as 36,000 different species^[See @frank_molecular-phylogenetic_2007 or click for the [open access download](http://www.pnas.org/content/104/34/13780.full)]!  With that much variety, how do we find the ones that matter?

Fortunately, only about 14 strains of 10 species account for 80% of a typical gut microbiome^[See the detailed estimates here: @kraal_prevalence_2014]

In this chapter, we'll just consider the most common microbes and the overall consensus on what they do.  Later, in the chapter on [experiments](#experiments), we'll show more about how you can _manipulate_ them.

What you're really wondering is how does your sample compare to others? Do you have an unusual abundance (or lack) of a particular taxa? Is there something that might indicate a greater or lesser similarity between your sample and certain other types of people? That is a very difficult question which we'll address over and over in this book, but for now let's just look at overall abundances of some key microbes. 

## Phylum {#microbesPhylum}

This section is under construction

The gut microbiome of most westerners is dominated by _Firmicutes_ and _Bacteroidetes_, which generally make up 80% or more of the total sample.  Most people also have smaller amounts of _Actinobacteria_,  _Proteobacteria_ and _Verrucomicrobia_. This overall composition is so common in healthy people that it's tempting to assume their dominance is "natural" or "normal", but like much else with the microbiome, the situation is different outside the western world, a clue that it's difficult to summarize a single individual's microbiome as "good" or "bad." It all depends.



## Genus {#microbesGenus}

This section is under construction

You're likely to hear most about the genus level because it's the most detail that cheap sequencing technologies can get right -- most of the time. 


_Bifidobacterium_ is a key component of virtually all popular probiotic supplements, partly because it is so easy to manufacture, but also due to its proven association with sleep and other aspects of health. A six month picture of my levels shows some dramatic ups and downs

<!-- ```{r summarPlotBifido} -->
<!-- #mhg_taxa_plot(subset_samples(gut.norm, Date > "2016-06-01"),"Akkermansia") -->

<!-- probiotic.day <- seq(as.Date("2016-10-23"),length=9, by=1) -->
<!-- plot_bar(subset_taxa(subset_samples(gut.norm, Date %in% seq(as.Date("2016-07-01"), length = 180, by = 1)), -->
<!--                      Genus == "Bifidobacterium"), x = "Date") +  -->
<!--   geom_smooth(method = "loess", se = FALSE) + -->
<!--   annotate("point",x=probiotic.day, y = -0.03, color = "red")+ -->
<!--   scale_y_continuous(labels = function(x) x / 10000) + ylab("Abundance (%)") + ggtitle("Bifidobacterium") -->

<!-- ``` -->

<!-- Incidentally, the red dots indicate days when I was taking a powerful probiotic supplement that contained _Bifidobacterium_. And that big spike in September? That was during a trip to New Orleans, when I ate a lot of red beans and rice.  At least for me, food seems to work better than taking supplements. -->

## Species

This section is under construction

When you hear the term "species", you probably think of a specific kind of creature, like a dog or a cat. More generally, among the kinds of plants and animals we encounter in the visible world, the term "species" refers broadly to organisms that can mate with one another to produce offspring of the same kind. Cats and dogs are different species because they can't mate with each other.

But bacteria don't mate: they reproduce by dividing themselves in half. So how do we define a species? In fact, even terms like "parent" or "child" aren't quite appropriate if each new cell is an identical copy of the old one. For very broad categories, like phylum or even genus, the similarities among like cells is high enough that we feel comfortable grouping them together with a common name, but at what point do we reach the lowest, most _specific_ level. 

The answer is tricky for another reason, called _horizontal gene transfer_, a process by which sometimes (in fact, quite often), a microbe will absorb genes from nearby microbes, altering its genome and its corresponding functions, sometimes significantly. Once that happens, the resulting new microbe can itself divide indefinitely, producing more and more copies of itself with the new gene. Although the new microbes still mostly resemble their original 

<!-- ## Diversity -->

<!-- The general consensus is that diversity is good: a greater variety of microbes ensures more resillience against the daily threat of invaders. Many people, after taking just one test, often feel either reassured that their diversity is "good" or disappointed that it's "bad". But I find that day-to-day variability is great enough that it's almost never useful to use a single result. For example, here's my diversity during a typical week: -->


<!-- ```{r, plotOneWeekDiversity, echo= FALSE, warning = FALSE} -->
<!-- plot_richness(subset_samples(sprague.family.norm, Date > "2017-03-15" & Date < "2017-03-27" & Site == "gut"), measures=c("InvSimpson"),x = "Date") + -->
<!--   scale_x_date(date_labels = "%m-%d (%A)", date_breaks = "1 day", date_minor_breaks = "1 day") + -->
<!--   geom_smooth(method = "loess") +  -->
<!--   ggtitle("Gut diversity changes day-to-day") -->
<!-- ``` -->

<!-- If I had only tested on Monday, I may have been disappointed with my 0.38 score. Wait another day or two and, with no significant changes in diet, I was up to 0.72 -- before plunging to 0.37 by the weekend. Moral: don't take a single result too seriously. -->

<!-- To get a sense of how much diversity can vary over a year: -->

<!-- ```{r, plotOneYearDiversity, echo= FALSE, warning = FALSE} -->
<!-- plot_richness(subset_samples(sprague.family.norm, Date > "2016-06-01" & Site == "gut" & Reads > 10000), measures=c("InvSimpson"), x = "Date") + -->
<!--   scale_x_date(date_labels = "%m-%Y", date_breaks = "3 months", date_minor_breaks = "1 month") + -->
<!--   geom_smooth(method = "loess") +  -->
<!--   ggtitle(paste("Gut diversity year-to-year through", -->
<!--                 tail(sample_data(gut.norm)$Date,1))) -->

<!-- ``` -->

<!-- Although the blue moving-average line shows apparent stability, there are many days that are far above and below the average. Yes, at various times during this period I was eating different types of food, often in a deliberate attempt to influence my microbiome, but believe me: that is not the reason for the wild changes up and down. I also studied the mathematics behind how diversity is measured, hoping to find something more "accurate", but ultimately I concluded that, like many attempts to summarize the microbiome in a single number, the whole concept of diversity is a mirage. Everything depends. -->




